virttest/utils_net.py

```diff
@@ -1045,9 +1046,16 @@ def get_structure(self):
         result = dict()
         for br_iface in os.listdir(sysfs_path):
             br_iface_path = os.path.join(sysfs_path, br_iface)
-            if (not os.path.isdir(br_iface_path) or
-                    "bridge" not in os.listdir(br_iface_path)):
-                continue
+            if os.path.isdir(br_iface_path):
+                try:
+                    if "bridge" not in os.listdir(br_iface_path):
+                        continue
+                except OSError as e:
+                    if e.errno == errno.ENOENT:
+                        continue
+                    else:
+                        raise e
+
             result[br_iface] = dict()
```

Comment:
@PaulYuuu: The block after this change will execute if `os.path.isdir(br_iface_path) == False` with your change and will not execute before this change took place. Is this intentional?

We encounter similar errors in the block below possibly due to the same reasons like:

```
20:17:00 ERROR|   File "/usr/lib/python3.7/site-packages/virttest/utils_net.py", line 1100, in port_to_br
20:17:00 ERROR|     if port_name in self.get_structure()[br]['iface']:
20:17:00 ERROR|   File "/usr/lib/python3.7/site-packages/virttest/utils_net.py", line 1065, in get_structure
20:17:00 ERROR|     with open(stp_state_path, "r") as stp_state_file:
20:17:00 ERROR| FileNotFoundError: [Errno 2] No such file or directory: '/sys/class/net/t0-W6kgkf/bridge/stp_state'
```

and sometimes

```

2019-07-09 16:27:01,051 stacktrace       L0045 ERROR|   File "/usr/lib/python3.7/site-packages/virttest/utils_net.py", line 1100, in port_to_br
2019-07-09 16:27:01,051 stacktrace       L0045 ERROR|     if port_name in self.get_structure()[br]['iface']:
2019-07-09 16:27:01,051 stacktrace       L0045 ERROR|   File "/usr/lib/python3.7/site-packages/virttest/utils_net.py", line 1066, in get_structure
2019-07-09 16:27:01,052 stacktrace       L0045 ERROR|     stp_state = int(stp_state_file.read().strip())
2019-07-09 16:27:01,052 stacktrace       L0045 ERROR| OSError: [Errno 9] Bad file descriptor
```

Can we put this entire part in a try block instead?

Reply:
@pevogam Sorry for my fault. Could you take a time to review #2111 ? I think it can solve the above problem.

Reply:
Ok, I will take a look although I don't use Open vSwitch so I can only comment on the parts related to this :)